kratos/python_scripts/analysis_stage.py

```diff
@@ -411,6 +416,96 @@ def __CreateListOfProcesses(self):
         self._list_of_processes.extend(self._list_of_output_processes) # Adding the output processes to the regular processes
         self._list_of_output_processes.extend(deprecated_output_processes)
 
+    def __AddProcessesVariables(self):
+        variables_list = []
+        if self.project_parameters.Has("processes"): #TODO: Here we are assuming that output_processes cannot add variables. Decide this in KTC.

```

Comment:
Is this final? I wouldn't forbid output processes from requiring variables. For example, visualizing multifreedom constraints #1225 needs dummy variables to store the coefficients and constraint gaps that can be loaded in the postprocessing software.

This could be useful for other debugging applications as well, like writing the IDs of nodes/elements/conditions, or any other quantity that is not usually stored in variables.

Variables used for these purposes would probably be ones that aren't otherwise used during the simulation, so they'd have to be loaded separately.

Reply:
Nope, it is not final. Indeed, you raised a very fair observation that makes perfect sense to me.